google/cloud/storage/internal/grpc_bucket_name.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_GRPC_BUCKET_NAME_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_GRPC_BUCKET_NAME_H
+
+#include "google/cloud/storage/version.h"
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+std::string BucketNameToProto(std::string const& bucket_name);
+std::string BucketNameFromProto(std::string const& bucket_name);
```

Comment:
I'm not sure "proto" enters into this.  Would these be better as something like `BucketNameToId(bucket_name)` and `BucketIdToName(bucket_id)`?

Reply:
I think the name should hint that it is gRPC related, as the `id` and `name` are identical for REST.  Please let me know if the new names make sense.